### PR TITLE
Added stm32l476 support, trivial source code cleanup

### PR DESCRIPTION
**Pull request by Kochise**
Source code cleanup made for static analyzers

### CODE REVIEW

**Comment by grapherd** [comment]
@Kochise Thanks for the porting to stm32L476, but it seems broke the build of the exists board on CI and locals (which I build on my computer and raise the error, too).

Could you help to check the build error, and fix it?
Thanks.

**Comment by Kochise** [comment]
Will check that this week-end and provide a makeit script for it. Any compiler log so far to point out where the trouble is ?

**Comment by grapherd** [comment]
@Kochise you can found it at travis-ci [here](https://travis-ci.org/f9micro/f9-kernel/builds/217574958).

I'm not sure why `ipc.c` is affect by your code, but three of the exists board all were failed.

**Comment by Kochise** [comment]
ipc.c, line 251, remove the badly formatted line :
>>>				dbg_printf(DL_IPC,
				dbg_printf(DL_IPC, "IPC: %t thread start\n", to_tid);

**Comment by Kochise** [comment]
Also forgot to modify original build scripts to include the new board/platform, sorry, will commit that asap.

File: board/nucleol476rg/board.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/* Copyright (c) 2016 The F9 Microkernel Project. All rights reserved.
```
Comment (jserv) [inline board/nucleol476rg/board.c]: You can bump the date to `2017` since your contributions show up.

File: kernel/interrupt.c [hunk 2]
```diff
@@ -74,10 +74,11 @@ static void user_irq_queue_push(struct user_irq *uirq)
 
 static struct user_irq *user_irq_queue_pop(void)
 {
+	struct user_irq *uirq;
 	if (user_irq_queue_is_empty())
 		return NULL;
 
-	struct user_irq *uirq = user_irq_queue.head;
+	uirq = user_irq_queue.head;
```
Comment (jserv) [inline kernel/interrupt.c]: Can you split the changes against directory `kernel/` in another commit?

File: kernel/interrupt.c [hunk 3]
```diff
@@ -98,14 +101,17 @@ static void user_irq_queue_delete(int irq)
 
 static inline void user_irq_reset_all(void)
 {
-	for (int i = 0 ; i < IRQn_NUM ; i++)
+	int i;
+	for (i = 0 ; i < IRQn_NUM ; i++)
```
Comment (jserv) [inline kernel/interrupt.c]: Can't your compiler support C99 features like the statement `for (int i = 0 ; i < IRQn_NUM ; i++)`?

File: board/nucleol476rg/board.c (hunk 1 above)
Comment (Kochise) [inline board/nucleol476rg/board.c]: Will do that...

File: kernel/interrupt.c (hunk 2 above)
Comment (Kochise) [inline kernel/interrupt.c]: Will also do that, since I'm also moving sources (board, external, kernel, loader, platform, user) into a \src folder

**Comment by Kochise** [comment]
Hmmm, perhaps I should install a Linux VM with Travis to test my builds against regular toolchains before committing :/

**Comment by Kochise** [comment]
OK, got it to compile, now I have to revert my commits back to origin and recommit them in two batches, one for the stm32l476 support, one for the kernel code 'beautification'.

**Pull request closed without merging**